microcloud/test/suites/basic.sh

```diff
@@ -110,7 +110,7 @@ test_instances_config() {
 
   # Setup a MicroCloud with 3 systems, ZFS storage, and a FAN network.
   addr=$(lxc ls micro01 -f csv -c4 | grep enp5s0 | cut -d' ' -f1)
-  lxc exec micro01 -- TEST_CONSOLE=0 microcloud init --preseed << EOF
+  lxc exec micro01 --env TEST_CONSOLE=0 -- microcloud init --preseed << EOF
```

Comment:
In this specific case, the command replaced does seem to be valid as LXD tries to `execve` `TEST_CONSOLE=0` which is not a valid binary in the instance. Here's a simple reproducer:
```
$ lxc exec c1 -- FOO=1 env
Error: Command not found

$ lxc exec c1 --env FOO=1 -- env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin
container=lxc
FOO=1
TERM=xterm-256color
HOME=/root
USER=root
LANG=C.UTF-8
```

There are many more examples where I replaced `lxc exec foo -- sh -c " ... "` to drop the wrapper shell mostly because I find it opens the door to doing a lot inside that subshell and it's easy to lose visibility (forgetting to use `sh -xc` for example). Mostly a question of style/taste I guess.

I've kept the `sh -c` where it was needed for IO redirections when the receiving file (usually `> out`) was meant to live inside the instance running the command.

Reply:
In general when we see `sh -c` being used its usually a mistake.

Reply:
I see, thanks for the detailed explanation!